jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1567,7 +1567,24 @@ Method getMethodHelper(
 			boolean scanInterfaces = false;
 			for (Method m: methods) {
 				MethodInfo mi = new MethodInfo(m);
-				myMethods.put(mi, mi);
+				MethodInfo prevMI = myMethods.put(mi, mi);
+				if ( prevMI != null) {
+					/* As per Java spec:
+					 * For methods with same signature (name, parameter types) and return type,
+					 * only the most specific method should be selected. 
+					 * Method N is more specific than M if:
+					 * N is declared by a class and M is declared by an interface; or
+					 * N and M are both declared by either classes or interfaces and N's
+					 * declaring type is the same as or a subtype of M's declaring type.
+					 */
+					Class<?> miDeclaringClz = mi.me.getDeclaringClass();
```

Comment:
Can we call this `mDeclaringClass` and  fetch it after the `mi` is created?  Line 1588 immediately fetches the same value so looking it up once would be better.

Reply:
Agreed, fetching `mDeclaringClass` once and use it two cases.